### PR TITLE
Remove support for EKS 1.17

### LINKED ISSUE #4485: Remove support for 1.17
EKS deprecated 1.17, you can no longer provision them. This is causing some of our integration tests to fail. lets remove support as we did for 1.16 https://github.com/weaveworks/eksctl/pull/3944

### PR DESCRIPTION
**Pull request by Himangini**
### Description
Closes #4485 

```
eksctl create cluster --version 1.17                                                                                         1 ↵
2021-11-24 17:14:31 [ℹ]  eksctl version 0.76.0-dev+020d876f.2021-11-24T16:39:14Z
2021-11-24 17:14:31 [ℹ]  using region us-east-1
Error: invalid version, 1.17 is no longer supported, supported values: 1.18, 1.19, 1.20, 1.21
see also: https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html
```

### Checklist
- [ ] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/addons/default/coredns_test.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ var _ = Describe("default addons - coredns", func() {
 
 		Context("when CoreDNS is NOT up to date", func() {
 			BeforeEach(func() {
-				input.ControlPlaneVersion = "1.18.x"
+				input.ControlPlaneVersion = "1.19.x"
```
Comment (Skarlso) [inline pkg/addons/default/coredns_test.go:83]: Huh, was this meant to be updated?

Comment (Skarlso) [reply pkg/addons/default/coredns_test.go:83]: Oh, I see why :)

**Review by cPu1 — COMMENTED** [review]
[userdocs/src/introduction.md](https://github.com/weaveworks/eksctl/blob/main/userdocs/src/introduction.md?plain=1#L26) needs to be updated to remove mention of 1.16 and 1.17.

**Comment by Himangini** [comment]
> [userdocs/src/introduction.md](https://github.com/weaveworks/eksctl/blob/main/userdocs/src/introduction.md?plain=1#L26) needs to be updated to remove mention of 1.16 and 1.17.

Nice spot @cPu1 I've removed it 👍🏻

**Review by cPu1 — APPROVED** [review]

**Pull request merged**